### ISSUE TITLE
[input_as_map] Use NewFieldPath(....).Name() for Fieldnames in modelgen.go

### ISSUE THREAD

**Issue by DENKweit** [infiotinc/gqlgenc#13]
In clientgen.go:
```diff
- genType := r.GetGenType(def.Name)
+ genType := r.GetGenType(NewFieldPath(def.Kind, def.Name).Name())
```

**Comment by raphaelfff** [comment]
There is no file called `clientgen.go`, can you be more precise what is the issue here ?

**Comment by DENKweit** [comment]
Sorry, I meant ` clientgen/modelgen.go` 

The issue is that otherwise genType might be `nil` because e.g. def.name is `FooType_hash` instead of `FooTypeHash`

**Comment by raphaelfff** [comment]
Fixed in a1bc669
